commitlint/plugins.ts

```diff
@@ -399,7 +399,15 @@ export abstract class Plugins {
 
                         let lineIsFooterNote = Helpers.isFooterNote(line);
 
-                        if (!isUrl && !lineIsFooterNote) {
+                        let index = line.search(`([0-9a-f]{40})[\.\,\:\;\?\!]?$`);
```

Comment:
@webwarrior-ws let's split this pattern in two variables please, to make it more readable, one called commitHashPattern and the other anySinglePunctuationCharOrNothing

Reply:
Split into 2 variables.